tests/ytt/external_blobstore_test.go

```diff
@@ -1,30 +1,48 @@
 package ytt
 
 import (
+	"io/ioutil"
+	"os"
+
 	. "code.cloudfoundry.org/yttk8smatchers/matchers"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 )
 
 var _ = Describe("External Blobstore", func() {
-
 	var ctx RenderingContext
 	var data map[string]interface{}
-	var templates []string
+	var templateFiles []string
+	var valueFiles []string
+	var targetDir string
+	var err error
 
 	BeforeEach(func() {
-		templates = []string{
+		templateFiles = []string{
 			pathToFile("config/minio"),
+		}
+
+		valueFiles = []string{
 			pathToFile("tests/ytt/blobstore/blobstore-values.yml"),
 		}
 	})
 
 	JustBeforeEach(func() {
-		ctx = NewRenderingContext(templates...).WithData(data)
+		targetDir, err = ioutil.TempDir("", "")
+		Expect(err).NotTo(HaveOccurred())
+
+		ctx = NewRenderingContext(targetDir, valueFiles...).WithData(data)
+
+		err = ctx.CopyTemplatesToTargetDir(templateFiles...)
```

Comment:
Hmmm...purely a suggestion and probably a nit pick but I wonder if a slightly better API might be?

```
ctx = NewRenderingContext(targetDir, valueFiles...).WithData(data).WithTemplateFiles(templateFiles...)
```

And just have `WithTemplateFiles` panic if it encounters an error.  Or have a final build() call that spits out and previous error? 

Might be slightly better?  I always find methods like `CopyTemplatesToTargetDir ` confusing because the target dir was previously given to the RenderingContext in its "constructor".

Reply:
Yeah I know what you mean. I don't like the idea of intentional panics in code as that would break the tests as opposed to causing them to fail. The `build()` call to check for errors sounds possible, but it seems fragile in that tests could forget to call it.

Reply:
Builder pattern is too Java-centric too.  

I think what we need is the [functional options](https://www.calhoun.io/using-functional-options-instead-of-method-chaining-in-go/) pattern.  I have seen this a few times in our code (Credhub is the one I remember).  Perhaps something like this would work better?

```
type RenderingContextOption func(ctx Context) (Context, error)

func NewRenderingContextWithOptions(string, RenderingContextOption...) (Context, error)

ctx, err := NewRenderingContextWithOptions(targetDir,
  WithData(data),
  WithTemplates(templateFiles),
)
Expect(err).NotTo(HaveOccurred())
```